### PR TITLE
fix: return disabled variant when feature evaluated false

### LINKED ISSUE #171: Bug: when feature is evaluated false, it should return disabled variant
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Currently, when feature is evaluated false, it is still returning a variant from feature, but should return disabled variant.
Also added client specification test for it https://github.com/Unleash/client-specification/pull/57

### To reproduce

1. Update client-specification to 4.2.3 and run tests, which will fail

### Sample code (optional)

_No response_

### Version

1.12.1

### Expected behavior

Feature should return disabled variant, when feature is evaluated false

### Logs (optional)

_No response_

### Additional context (optional)

I am working on the fix, just putting up issue for documentation

### PR DESCRIPTION
**Pull request by sjaanus**
Fixes https://github.com/Unleash/unleash-client-php/issues/171

1. Updated client spec to 4.2.3
2. Now getVariant will return disabled variant when feature is evaluated to false.

### CODE REVIEW

File: src/DefaultUnleash.php [hunk 1]
```diff
@@ -39,16 +40,60 @@ public function __construct(
...
+        $feature = $this->findFeature($featureName);
+        return $this->isFeatureEnabled($feature, $default);
+    }
+
+    public function getVariant(string $featureName, ?Context $context = null, ?Variant $fallbackVariant = null): Variant
+    {
+        $fallbackVariant ??= $this->variantHandler->getDefaultVariant();
         $context ??= $this->configuration->getContextProvider()->getContext();
 
+        $feature = $this->findFeature($featureName);
```
Comment (kwasniew) [inline src/DefaultUnleash.php]: same as lines 43 and 44? Can we call isEnabled?

File: src/DefaultUnleash.php [hunk 2]
```diff
@@ -39,16 +40,60 @@ public function __construct(
 
     public function isEnabled(string $featureName, ?Context $context = null, bool $default = false): bool
```
Comment (kwasniew) [inline src/DefaultUnleash.php:41]: why is context passed if we don't use it?

**Review by sighphyre — APPROVED** [review]
I think this looks good

**Comment by sjaanus** [comment]
Codacy review seems not relevant.

**Pull request merged**